apis/apps/v1alpha1/cluster_types.go

```diff
@@ -92,6 +93,41 @@ type ClusterSpec struct {
 	// customized parameters that is used in different clusterdefinition
 	// +optional
 	Parameters map[string]string `json:"parameters,omitempty"`
+
+	// cluster backup configuration.
+	// +optional
+	Backup *ClusterBackup `json:"backup,omitempty"`
+}
+
+type ClusterBackup struct {
+	// enabled defines whether to enable automated backup.
+	// +kubebuilder:default=false
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// retentionPeriod is a time string ending with the 'd'|'D'|'h'|'H' character to describe how long
+	// the Backup should be retained. if not set, will be retained forever.
+	// +kubebuilder:validation:Pattern:=`^\d+[d|D|h|H]$`
+	// +kubebuilder:default="1d"
+	// +optional
+	RetentionPeriod *string `json:"retentionPeriod,omitempty"`
+
+	// backup method, support: snapshot, backupTool.
+	// +kubebuilder:validation:Enum=snapshot;backupTool
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=backupTool
+	Method dataprotectionv1alpha1.BackupMethod `json:"method"`
+
+	// the cron expression for schedule, the timezone is in UTC. see https://en.wikipedia.org/wiki/Cron.
+	// +optional
+	CronExpression string `json:"cronExpression,omitempty"`
+
+	// startWindowMinutes defines the time window for starting the job if it misses scheduled
+	// time for any reason. the unit of time is minute.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1440
+	StartWindowMinutes *int64 `json:"startWindowMinutes,omitempty"`
 }
```

Comment:
Perhaps add an optional `repoName` field to allow the user to specify a backup repo.

Reply:
Added. Check it plz.

Reply:
`backupPolicy.spec.{data,log}file.backupRepoName` should be updated correspondingly in mergeClusterBackup().